receipts_test.go

```diff
@@ -0,0 +1,175 @@
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/0xsequence/ethkit/go-ethereum/common"
+	"github.com/0xsequence/ethkit/go-ethereum/core/types"
+	"github.com/0xsequence/go-sequence"
+	"github.com/0xsequence/go-sequence/testutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReceiptDecoding(t *testing.T) {
+	// Ensure a dummy sequence wallet is deployed
+	wallets, err := testChain.DummySequenceWallets(1, 1)
+	assert.NoError(t, err)
+	assert.NotNil(t, wallets)
+	assert.Equal(t, len(wallets), 1)
+	wallet := wallets[0]
+
+	// Create normal txn of: callmockContract.mockMint(wallet, 10)
+	callmockContract, _ := testChain.Deploy(t, "ERC20Mock")
+	calldata, err := callmockContract.Encode("mockMint", wallet.Address(), big.NewInt(10))
+	assert.NoError(t, err)
+
+	// Sign and send the transaction
+	err = testutil.SignAndSend(t, wallet, callmockContract.Address, calldata)
+	assert.NoError(t, err)
+
+	var send [][]byte
+	for i := 0; i <= 10; i++ {
+		data, err := callmockContract.Encode("transfer", common.Address{1}, big.NewInt(int64(i)))
+		assert.NoError(t, err)
+		send = append(send, data)
+	}
+
+	// These are wallet's intended transactions
+	bundle := sequence.Transactions{
+		{
+			To:   callmockContract.Address,
+			Data: send[1],
+		}, // logs: 3 (1 from NonceChange), balance: 9, executed
+		{
+			To:   callmockContract.Address,
+			Data: send[1],
+		}, // logs: 5, balance: 8, executed
+		{
+			To: wallet.Address(),
+			Transactions: sequence.Transactions{
+				{
+					To:   callmockContract.Address,
+					Data: send[2],
+				}, // logs: 7, balance: 6, executed
+				{
+					To:   callmockContract.Address,
+					Data: send[10],
+				}, // logs: 8, balance: -4, failed
+			},
+		}, // logs: 9, balance: 6, executed
+		{
+			To: wallet.Address(),
+			Transactions: sequence.Transactions{
+				{
+					To:            callmockContract.Address,
+					Data:          send[3],
+					RevertOnError: true,
+				}, // logs: 11, balance: 3, reverted
+				{
+					To:            callmockContract.Address,
+					Data:          send[10],
+					RevertOnError: true,
+				}, // logs: 12, balance: -7, reverted
+			},
+		}, // logs: 10, balance: 6, failed
+		{
+			To:   callmockContract.Address,
+			Data: send[10],
+		}, // logs: 11, balance: -4, failed
+		{
+			To:   callmockContract.Address,
+			Data: send[4],
+		}, // logs: 13, balance: 2, executed
+		{
+			To:   callmockContract.Address,
+			Data: send[4],
+		}, // logs: 14, balance: -2, failed
+		{
+			To:   callmockContract.Address,
+			Data: send[2],
+		}, // logs: 16, balance: 0, executed
+	}
+
+	signedBundle, err := wallet.SignTransactions(context.Background(), bundle)
+	assert.NoError(t, err)
+
+	_, _, waitReceipt, err := wallet.SendTransaction(context.Background(), signedBundle)
+	assert.NoError(t, err)
+	receipt, err := waitReceipt(context.Background())
+	assert.NoError(t, err)
+	assert.Equal(t, len(receipt.Logs), 16)
+
+	receipts, err := sequence.DecodeReceipt(receipt, signedBundle)
```

Comment:
is it possible for use to `DecodeReceipt` just with a *types.Receipt object itself? ..without the signedBundle? the idea is a client will decode the receipt, without actually having access to the signedBundle.. it could fetch the "data" from the transaction and parse it, thats all it would have